File: AdminServer/AdminServerApiClient/src/main/java/com/rkarp/adminserver/dao/UserDAO.java
package com.rkarp.adminserver.dao;

import java.util.List;

import com.rkarp.adminserver.exception.DuplicateUserException;
import com.rkarp.adminserver.exception.UserNotFoundException;
import com.rkarp.adminserver.model.User;

public interface UserDAO {

    public void addUser(User user) throws DuplicateUserException;

    public User getUser(int userId) throws UserNotFoundException;
    
    public User getUser(String username) throws UserNotFoundException;

    public void updateUser(User user) throws UserNotFoundException, DuplicateUserException;

    public void deleteUser(int userId) throws UserNotFoundException;

    public List<User> getUsers();

}

File: Foundation/src/main/java/com/rkarp/foundation/dao/UserDAO.kt
package com.rkarp.foundation.dao

import com.rkarp.foundation.model.Credentials
import com.rkarp.foundation.model.User

interface UserDAO {
    fun getUserFromToken(token: String): User?
    fun getUserByName(username: String): User?
    fun updateUserToken(user: User, token: String)
    fun registerUser(user: Credentials): User?
}
